client/src/main/java/com/aliyun/emr/rss/client/compress/RssZstdCompressor.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.aliyun.emr.rss.client.compress;
+
+import java.util.zip.CRC32;
+import java.util.zip.Checksum;
+
+import com.github.luben.zstd.Zstd;
+
+public class RssZstdCompressor extends RssZstdTrait implements Compressor {
+  private final int compressionLevel;
+  private final Checksum checksum;
+  private byte[] compressedBuffer;
+  private int compressedTotalSize;
+
+  public RssZstdCompressor() {
+    this(256 * 1024);
+  }
+
+  public RssZstdCompressor(int blockSize) {
+    this(blockSize, 1);
+  }
+
+  public RssZstdCompressor(int blockSize, int level) {
+    compressionLevel = level;
+    checksum = new CRC32();
+    initCompressBuffer(blockSize);
+  }
+
+  @Override
+  public void initCompressBuffer(int maxDestLength) {
+    int compressedBlockSize = HEADER_LENGTH + maxDestLength;
+    compressedBuffer = new byte[compressedBlockSize];
+    System.arraycopy(MAGIC, 0, compressedBuffer, 0, MAGIC_LENGTH);
+  }
+
+  @Override
+  public void compress(byte[] data, int offset, int length) {
+    checksum.reset();
+    checksum.update(data, offset, length);
+    final int check = (int) checksum.getValue();
+    int maxDestLength = (int) Zstd.compressBound(length);
```

Comment:
Should change to use long?

Reply:
I don't think so since the maximum size of an array in java is Int.MaxValue(2G).